lib/matplotlib/backends/_backend_tk.py

```diff
@@ -882,5 +883,39 @@ def trigger_manager_draw(manager):
     @staticmethod
     def mainloop():
         managers = Gcf.get_all_fig_managers()
-        if managers:
+        if not managers:
+            return
+        if not is_tk_mainloop_running():
             managers[0].window.mainloop()
+
+
+def is_tk_mainloop_running():
```

Comment:
how does this compare with https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/cbook/__init__.py#L65 (possibly combined with checking that the thread matches `threading.main_thread()`)?

Reply:
In the contexts that I call `is_tk_mainloop_running`, `_get_running_interactive_framework` returns `None`, not "tk".  This is because `_get_running_interactive_framework` is incorrectly implemented. There are several mainloop functions in various classes in tkinter and it only checks for one of them. Importantly one of them is a built-in method that has no `__code__` attribute to check ("built-in method mainloop of _tkinter.tkapp object") so i'm not sure how to check for it.

Reply:
I see your point.  Adding a check for Misc.mainloop as you did certainly seems better.  I don't have a good solution for the general case, though.